lib/crypto/hkdf.lua

```diff
@@ -0,0 +1,115 @@
+--
+-- SPDX-FileCopyrightText: (c) 2025 jperon <cataclop@hotmail.com>
+-- SPDX-License-Identifier: MIT OR GPL-2.0-only
+--
+
+--- HMAC-based Extract-and-Expand Key Derivation Function (HKDF) based on RFC 5869.
+-- This module provides functions to perform HKDF operations, utilizing the
+-- underlying `crypto_shash` C module for HMAC calculations.
+-- @classmod crypto.hkdf
+
+local shash = require("crypto_shash")
+local char, pack, rep, sub = string.char, string.pack, string.rep, string.sub
+
+--- HKDF operations.
+-- This table provides the `new` method to create HKDF instances and also
+-- serves as the prototype for these instances.
+local HKDF = {}
+
+function HKDF.__len(self)
+	return self.tfm:digestsize()
+end
+
+--- Creates a new HKDF instance for a given hash algorithm.
+-- @function HKDF.new
+-- @tparam string alg The base hash algorithm name (e.g., "sha256", "sha512").
+-- The "hmac(" prefix will be added automatically.
+-- @treturn HKDF An HKDF instance table with methods for key derivation.
+-- @usage local hkdf_sha256 = require("crypto.hkdf").new("sha256")
+function HKDF.new(alg)
+	local hmac = setmetatable({}, HKDF)
+	local ok, tfm = pcall(shash.new, "hmac(" .. alg .. ")")
+	assert(ok, "Invalid hash algorithm: " .. alg)
```

Comment:
sorry, I think you misunderstood my point.. I was just asking if it would be safe to pass an invalid alg to shash.new.. it seems so, thus you don't need this assert, just let shash.new to raise this error..